pytensor/link/numba/dispatch/elemwise_codegen.py

```diff
@@ -1,3 +1,5 @@
+from __future__ import annotations
```

Comment:
Why is this needed?

Reply:
Otherwise in Python 3.8 you get `pytensor/link/numba/dispatch/elemwise_codegen.py:72: error: "list" is not subscriptable, use "typing.List" instead`

Reply:
I'm okay either way, but that's why I changed it to run pre-commit in all supported versions. We missed this issue because it was running only in 3.9